cmd/process-agent/collector.go

```diff
@@ -106,6 +108,34 @@ func isRealTimeCheck(checkName string) bool {
 
 // NewCollectorWithChecks creates a new Collector
 func NewCollectorWithChecks(cfg *config.AgentConfig, checks []checks.Check, runRealTime bool) Collector {
+	queueSize := ddconfig.Datadog.GetInt("process_config.queue_size")
+	if queueSize <= 0 {
+		log.Warnf("Invalid check queue size: %d. Using default value: %d", queueSize, ddconfig.DefaultCheckQueueSize)
+		queueSize = ddconfig.DefaultCheckQueueSize
+	}
+
+	rtQueueSize := ddconfig.Datadog.GetInt("process_config.rt_queue_size")
+	if rtQueueSize <= 0 {
+		log.Warnf("Invalid rt check queue size: %d. Using default value: %d", rtQueueSize, ddconfig.DefaultRTCheckQueueSize)
+		rtQueueSize = ddconfig.DefaultRTCheckQueueSize
+	}
+
+	queueBytes := ddconfig.Datadog.GetInt("process_config.process_queue_bytes")
+	if queueBytes <= 0 {
+		log.Warnf("Invalid queue bytes size: %d. Using default value: %d", queueBytes, ddconfig.DefaultProcessQueueBytes)
+		queueBytes = ddconfig.DefaultProcessQueueBytes
+	}
+
+	processResults := api.NewWeightedQueue(queueSize, int64(queueBytes))
+	log.Debugf("Creating process check queue with max_size=%d and max_weight=%d", processResults.MaxSize(), processResults.MaxWeight())
+
+	// reuse main queue's ProcessQueueBytes because it's unlikely that it'll reach to that size in bytes, so we don't need a separate config for it
+	rtProcessResults := api.NewWeightedQueue(rtQueueSize, int64(queueBytes))
+	log.Debugf("Creating rt process check queue with max_size=%d and max_weight=%d", rtProcessResults.MaxSize(), rtProcessResults.MaxWeight())
+
+	podResults := api.NewWeightedQueue(queueSize, int64(cfg.Orchestrator.PodQueueBytes))
+	log.Debugf("Creating pod check queue with max_size=%d and max_weight=%d", podResults.MaxSize(), podResults.MaxWeight())
```

Comment:
I think this approach is good, as long as we are only using these variables in one place.

I think it would be a bit nice to be able to query the config for these variables like everything else, but since we're only using them in the collector, I dont think it's a major issue.

Reply:
I agree with you but we basically had two options:

A) Validate the settings while we load the config so they can be read without any further validation

Advantages:
* All validation is done in a single place leading to a cleaner solution 

Disadvantages:
* Validation is done separately from where the setting is actually used. Who is reading the code should be aware of the existence of a previous validation step
* If we use a setting at two different places and they have different pre-requisites, how would we deal with the early validation step?

B) Leave the `pkg/config` package deal with only reading the settings + populating default values and do the validation when we're reading them

Advantages:
* No need to be aware of a previous validation step
* Different parts of the code can set different requisites for settings values
* Validation logic is closer to who actually knows the business logic to deal with the values (process-agent `collector` in this case  and not the global `config` pkg)

Disadvantages:
* More verbose solution

For these reasons I decided to follow the approach B but I'm open to other opinions. @xornivore what do you think?

Reply:
I think this is fine, as long as @xornivore approves as well.

If we need to use something like `process_config.expvar_port` in multiple places, will we have to do validation every time it's accessed?

Reply:
Yes, we'll have to validate it every time we read the value. I'll work on the PR to move the remaining settings and we'll have a better idea on how it will look like!

Reply:
Honestly, I think this will actually be fine. If it becomes too big of a deal to validate settings, we can always create a helper for it. Nice work!